### PR TITLE
Moab cmake

### PR DESCRIPTION
**Pull request by gonuke**
This changes the build process for MOAB in the docker image to use CMake instead of Autotools.

This build system is more robust across different platforms and will be where ongoing improvements are focused.

depends on PR #684

### CODE REVIEW

File: news/PR-0684.rst [hunk 1]
```diff
@@ -0,0 +1,15 @@
+**Added:** None
+
+**Changed:** 
+* turn off fortran support in MOAB build
```
Comment (bam241) [inline news/PR-0684.rst:4]: fortran is disabled in #684

File: news/PR-0684.rst [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+* use '&&' to join successive build steps to fail build on first failure
+* remove specification of unused FORTRAN compiler for HDF5 & MOAB builds
+
+**Deprecated:** None
+
+**Removed:** None
+
+**Fixed:** 
+* remove bad flag for Eigen3 in MOAB build
```
Comment (bam241) [inline news/PR-0684.rst:13]: same this is done in #684

**Comment by bam241** [comment]
(sorry about my deleted comments, they were about the news file of the #684 PR...)

**Comment by gonuke** [comment]
Once this is ready, I have a draft branch that replaces LAPACK with Eigen3.....

**Comment by gonuke** [comment]
I think I messed up this branch/PR.  will close and open again.

**Pull request closed without merging**